modules/core/src/main/java/org/apache/ignite/internal/processors/configuration/distributed/DistributedEnumProperty.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2020 GridGain Systems, Inc. and Contributors.
+ *
+ * Licensed under the GridGain Community Edition License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.gridgain.com/products/software/community-edition/gridgain-community-edition-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.processors.configuration.distributed;
+
+import java.io.Serializable;
+import org.apache.ignite.IgniteCheckedException;
+import org.apache.ignite.internal.util.future.GridFutureAdapter;
+import org.apache.ignite.internal.util.typedef.internal.S;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Wrapper for any enumerator.
+ * It will be serialized by its originality into meta store.
+ *
+ * @param <T> Type of enum.
+ */
+public class DistributedEnumProperty<T extends Enum> implements DistributedChangeableProperty<T> {
+    /** This property stored enumerator as order. */
+    SimpleDistributedProperty<Integer> internal;
```

Comment:
Did you consider using instance name instead of ordinal value. Seems that instance name is more robust.

Reply:
Metastore would store a custom integer id, as it was recommended in meeting.